File: common/src/main/java/de/hype/bbsentials/shared/objects/minions/MinionStorage.java
package de.hype.bbsentials.shared.objects.minions;

public enum MinionStorage {
    SMALL("Small Storage", 3),
    MEDIUM("Medium Storage", 9),
    LARGE("Large Storage", 15),
    XLARGE("X-Large Storage", 21),
    XXLARGE("XX-Large Storage", 27);

    private final Integer storage;
    private final Integer storageSlots;
    private final String displayName;

    MinionStorage(String displayName, int slots) {
        this.displayName = displayName;
        this.storageSlots = slots;
        this.storage = slots * 64;
    }

    public Integer getStorage() {
        return storage;
    }

    public String getDisplayName() {
        return displayName;
    }

    public Integer getStorageSlots() {
        return storageSlots;
    }
}

File: common/src/main/java/de/hype/bingonet/shared/objects/minions/MinionStorage.kt
package de.hype.bingonet.shared.objects.minions

enum class MinionStorage(val displayName: String, slots: Int) {
    SMALL("Small Storage", 3),
    MEDIUM("Medium Storage", 9),
    LARGE("Large Storage", 15),
    XLARGE("X-Large Storage", 21),
    XXLARGE("XX-Large Storage", 27);

    val storage: Int
    val storageSlots: Int

    init {
        this.storageSlots = slots
        this.storage = slots * 64
    }
}
